### DISCUSSION TITLE
Chapter Questions

### DISCUSSION THREAD

**Discussion by Blushock** [enzo1982/freac#478] [Help & Support] [answered]
How does fre:ac know how to write chapters to files, particularly if you disable "Read chapters from files"? Also, do chapters apply to music files, or just audiobook?

**Comment by enzo1982** [comment, marked as answer]
When converting multiple input files/tracks to a single output file, the output file will contain a chapter per original source file/track. In other words, each joblist entry will become a chapter in the output file.

fre:ac does not transfer chapters contained in individual files when reading chapters is disabled. I.e. when you add a file containing multiple chapters to the joblist while the option to read chapters is disabled and then convert the file, the output file will not contain chapters.

When it comes to chapters, fre:ac does not make any difference between music and audio book files. Both can contain chapters.

**Comment by mrfragger** [comment]
Anyway to increase chapter limit?  Currently it's at 999.  I have an audiobook that has 1200 chapters. I guess only way currently is to split it up to 600 chapters and 600 chapters but not ideal.
![999](https://github.com/enzo1982/freac/assets/80211672/4bb4753b-8289-4dda-b19e-ac4bfb7855c3)

**Reply by enzo1982** [reply]
fre:ac has no limit on the number of chapters it can process, but some file/tagging formats have.

You are probably dealing with a file that uses the Vorbis Comment tagging format (used by FLAC, Vorbis, Opus and Speex). That format is limited to 999 chapters [as per the spec](https://wiki.xiph.org/Chapter_Extension#Format) (technically, 1000 chapters would be possible, but chapter numbers usually start at 1, so 000 is unused).

I assume the file you are trying to convert was created with a different program which limited the number of chapters to 999 to stay within the specification.

The limits for other tagging formats are:

- ID3v2: 255 chapters
- WMA: 65535 marker points
- MP4/M4A: 4 billion (so basically unlimited)
- Cue sheets (used with APEv2 tags): 99 tracks

While investigating this I found that fre:ac currently does not respect the maximum number of chapters for each format and in some cases writes invalid files when exceeding the limits. I'll fix that in an upcoming release and add warnings when trying to write files with more than the supported number of chapters.

**Reply by mrfragger** [reply]
wow great info..thanks a bunch.  Yes opus 16kbps chaptered audiobooks I make.  A couple years ago made tons of m4b aac 32kbps but no more.  So it's fine 999 limit I'll just split it when that's the case.  Also mpv has a 100 hour limit...in other words the chapters must appear 99  hours 59min or before to be seen.  So I can remember 1,000 chapters and 100 hour limit for opus chaptered audiobooks.
